### ISSUE TITLE
Invalid checksum error when attempting to read with a slice of length zero

### ISSUE THREAD

**Issue by lcdr** [mvdnes/zip-rs#138]
The below will error with "invalid checksum" at the `std::io::Read` step instead of simply reading nothing.

```rust
let src = std::fs::File::open(path).unwrap();
let mut zip = zip::ZipArchive::new(src).unwrap();

for i in 0..zip.len() {
    let mut file = zip.by_index(i).unwrap();
    let mut slice = [];
    std::io::Read::read(&mut file, &mut slice).unwrap();
}
```

**Comment by BenjaminRi** [comment]
This is already fixed by 3e5a8e6f980f2f1a5970670ddc1567423bee31cf, (#136).

**Comment by mvdnes** [comment]
I have just published a new version which includes the fix.
